From 6b108ec3be36bca5e5d81775932226a2ec83d74c Mon Sep 17 00:00:00 2001
From: Yoshisato Yanagisawa <yyanagisawa@chromium.org>
Date: Thu, 29 Feb 2024 07:21:12 +0000
Subject: [PATCH] Add crash key for calling CountFeature

To understand what feature to be recorded caused this, let me record the
feature ID in the crash key.

Bug: 40918057
Change-Id: I7a97f474e6d18cc7f929b2a0c591b746c7f33ef4
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5332953
Auto-Submit: Yoshisato Yanagisawa <yyanagisawa@chromium.org>
Reviewed-by: Minoru Chikamune <chikamune@chromium.org>
Reviewed-by: Shunya Shishido <sisidovski@chromium.org>
Commit-Queue: Yoshisato Yanagisawa <yyanagisawa@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1266862}
---

--- a/content/browser/service_worker/service_worker_container_host.cc
+++ b/content/browser/service_worker/service_worker_container_host.cc
@@ -9,6 +9,7 @@
 
 #include "base/containers/adapters.h"
 #include "base/containers/contains.h"
+#include "base/debug/crash_logging.h"
 #include "base/functional/callback_helpers.h"
 #include "base/functional/overloaded.h"
 #include "base/strings/stringprintf.h"
@@ -619,6 +620,7 @@
 void ServiceWorkerContainerHost::CountFeature(
     blink::mojom::WebFeature feature) {
   DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  SCOPED_CRASH_KEY_NUMBER("SWCH_CF", "feature", static_cast<int32_t>(feature));
 
   // CountFeature is a message about the client's controller. It should be sent
   // only for clients.
